### ISSUE TITLE
Add a "clean column names" option onimport

### ISSUE THREAD

**Issue by ettorerizza** [OpenRefine/OpenRefine#1401]
We may need to add a "clean column names" option in the import window. This would be particularly useful for importing Json files, which produce column names full of unnecessary characters.

![screenshot-127 0 0 1-3333-2017-12-27-11-15-39-018](https://user-images.githubusercontent.com/6286410/34378923-b6ad9368-eaf8-11e7-8789-3a60c8b7986b.png)

Having "clean" column names also allows us to use the simplified `cells.column_name.value` syntax rather than `cells ['Column Name'].value`

Resulting cleaned names should be unique and consist of only underscores (but not at the beginning of the name), ASCII lowercase letters and numbers.

**Comment by wetneb** [comment]
This is really needed. Are you thinking about a generic option that would be applicable to multiple importers? It seems to me that this issue is mostly focused on the JSON importer (and XML to some extent). So maybe adding an option to only these two could be enough.

**Comment by ettorerizza** [comment]
@wetneb Yes, I saw this as a generic option. It is almost essential for JSON files, but it would make it much easier to work in Open Refine, as I said above, as well as to export the file to a database.

Another possibility would be to allow a batch "transform" on the column names.

**Comment by thadguidry** [comment]
@wetneb @ettorerizza Yeah, we already have some issues for this at #1017 and #1241 and #898 
But I'll let you guys investigate those and come up with a broader brush in this issue to address all of these needs / issues.

**Comment by thadguidry** [comment]
Also #901 Rename multiple column names at once

**Comment by ostephens** [comment]
Additional support for this at https://groups.google.com/forum/#!topic/openrefine/ugbIcPJCBpM
